src/PageTypes/DatedUpdateHolderController.php

```diff
@@ -401,4 +408,13 @@ public function atom()
         );
         return $atom->outputToBrowser();
     }
+
+    /**
+     * The request has had an error that can result in a DB exception
+     * if filtering by the dates provided.
+     */
+    public function requestHasBadDates(): bool
```

Comment:
New API, would need to target `2` so doesn't work in terms of fixing the unit test on the previous branch

Reply:
In general for normal circumstances I'd agree. But I don't think this is going to cause any problems for anyone, and it will fix this bug.

I think we either need to accept that this is not how we normally treat semver, but since it won't break anything it's fine, _or_ target to 2 and merge there, and accept that anything earlier will not pass in CI.

Reply:
Unless you can think of a way to do this without adding the new method?
Would marking it `internal` help resolve any trepidation?